autogen/agentchat/contrib/functions/file_utils.py

```diff
@@ -0,0 +1,220 @@
+
+    Args:
+        file_path (str): The path to the PDF file.
+
+    Returns:
+        str: The extracted text from the PDF file.
+    """
+    import io
+    import requests
+    from pdfminer.high_level import PDFResourceManager, PDFPageInterpreter
+    from pdfminer.converter import TextConverter
+    from pdfminer.pdfpage import PDFPage
+
+    resource_manager = PDFResourceManager()
+    text_stream = io.StringIO()
+    converter = TextConverter(resource_manager, text_stream)
+    interpreter = PDFPageInterpreter(resource_manager, converter)
+
+    if file_path.startswith("http://") or file_path.startswith("https://"):
+        response = requests.get(file_path)
+        file = io.BytesIO(response.content)
+    else:
+        file = open(file_path, "rb")
+
+    for page in PDFPage.get_pages(file):
+        interpreter.process_page(page)
+
+    text = text_stream.getvalue()
+    converter.close()
+    text_stream.close()
+
+    return text
+
+
+@FunctionWithRequirements(python_packages=["python-docx"])
+def read_text_from_docx(file_path: str) -> str:
+    """
+    Reads text from a DOCX file and returns it as a string.
+
+    Args:
+        file_path (str): The path to the DOCX file.
+
+    Returns:
+        str: The extracted text from the DOCX file.
+    """
+    from docx import Document
+
+    doc = Document(file_path)
+    paragraphs = [p.text for p in doc.paragraphs]
+    text = "\n".join(paragraphs)
+
+    return text
+
+
+@FunctionWithRequirements(python_packages=["pillow", "requests", "easyocr"])
+def read_text_from_image(file_path: str) -> str:
+    """
+    Reads text from an image file or URL and returns it as a string.
+
+    Args:
+        file_path (str): The path to the image file or URL.
+
+    Returns:
+        str: The extracted text from the image file or URL.
+    """
+    import io
+    import requests
+    import easyocr
+    from PIL import Image
+
+    reader = easyocr.Reader(["en"])  # specify the language(s)
+
+    if file_path.startswith("http://") or file_path.startswith("https://"):
+        response = requests.get(file_path)
+        image = Image.open(io.BytesIO(response.content))
+    else:
+        image = Image.open(file_path)
+
+    output = reader.readtext(image)
+
+    # The output is a list of tuples, each containing the coordinates of the text and the text itself.
+    # We join all the text pieces together to get the final text.
+    text = " ".join([item[1] for item in output])
+
+    return text
+
+
+@FunctionWithRequirements(python_packages=["python-pptx"])
+def read_text_from_pptx(file_path: str) -> str:
+    """
+    Reads text from a PowerPoint file and returns it as a string.
+
+    Args:
+        file_path (str): The path to the PowerPoint file.
+
+    Returns:
+        str: The extracted text from the PowerPoint file.
+    """
+    from pptx import Presentation
+
+    presentation = Presentation(file_path)
+    text = ""
+
+    for slide in presentation.slides:
```

Comment:
Produced text that's all squished together with no spaces between text blocks, and no indication of what's on each slide.

I've modified this locally to:
```
    slide_num = 0
    for slide in presentation.slides:
        slide_num += 1

        text += f"\n\n<!-- Slide number: {slide_num} -->\n"

        for shape in slide.shapes:
            if shape.has_text_frame:
                text += shape.text + " "

        text = text.strip()
```